qamomile/udm/utils.py

```diff
@@ -0,0 +1,155 @@
+import networkx as nx
+import numpy as np
+from typing import List, Tuple, Set, Dict, Any, Union, Optional
+
+def simple_graph_from_edgelist(edgelist: List[Tuple[int, int]]) -> nx.Graph:
+    """Create a simple graph from an edge list."""
+    g = nx.Graph()
+    for i, j in edgelist:
+        g.add_edge(i, j)
+    return g
```

Comment:
The function simple_graph_from_edgelist can be simplified using:
```suggestion
def simple_graph_from_edgelist(edgelist: List[Tuple[int, int]]) -> nx.Graph:
    """Create a simple graph from an edge list."""
    g = nx.Graph()
    return nx.Graph(edgelist)
```